src/Components/Server/src/Builder/ComponentEndpointRouteBuilderExtensions.cs

```diff
@@ -18,7 +19,7 @@ public static class ComponentEndpointRouteBuilderExtensions
         /// </summary>
         /// <param name="endpoints">The <see cref="IEndpointRouteBuilder"/>.</param>
         /// <returns>The <see cref="IEndpointConventionBuilder"/>.</returns>
-        public static IEndpointConventionBuilder MapBlazorHub(this IEndpointRouteBuilder endpoints)
+        public static HubEndpointConventionBuilder MapBlazorHub(this IEndpointRouteBuilder endpoints)
```

Comment:
Shouldn't this be a different concrete type that derives from `HubEndpointConventionBuilder`?

Reply:
I didn't change it because the new builder type would be used by nothing.

We could make every framework return a strongly typed builder just in case: MVC, Razor pages, SignalR, Blazor, gRPC. Do we want to?

Reply:
Isn't that the point of this change? Future proofing? It's the same reason we want to avoid returning IEndpointConventionBuilder right?

Reply:
Couldn't you change MapBlazorHub and the like to return a more derived HubEndpointConventionBuilder later? Or would that be breaking? I don't think it would be since it's static.

Reply:
Pretty sure that’s a breaking change

Reply:
What code would break?

Reply:
Any compiled code would throw a MethodNotFoundException because the signature has changed

Reply:
That's what I was wondering. It's interesting the return type is used when looking up the method considering you cannot overload on return type.

Reply:
I guess it makes sense as it avoids potentially uglier errors if the return type changed to a non-derived type though.

Reply:
https://twitter.com/terrajobst/status/1115730579838787584

It is either the biggest co-incidence of all time or I think @davidfowl asked @terrajobst about it 😄

Reply:
> Isn't that the point of this change? Future proofing? It's the same reason we want to avoid returning IEndpointConventionBuilder right?

Ok. I agree it provides future proofing, I just wanted to double check that we've thought through changing to a concrete type for each framework.

Reply:
Yes, if we ever want extension methods that apply to components but not hubs we'll need it.

Reply:
We actually have an example of that already.

Reply:
As much as I would like to say it's unimportant because inheritance is bad.